apm-sniffer/apm-sdk-plugin/kafka-plugin/src/main/java/org/apache/skywalking/apm/plugin/kafka/CallbackInterceptor.java

```diff
@@ -16,42 +16,56 @@
  *
  */
 
-package org.apache.skywalking.apm.plugin.kafka.v1;
+package org.apache.skywalking.apm.plugin.kafka;
 
 import java.lang.reflect.Method;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
 import org.apache.skywalking.apm.agent.core.context.ContextManager;
 import org.apache.skywalking.apm.agent.core.context.ContextSnapshot;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
 import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
 import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
 
+/**
+ * @author zhang xin, stalary
+ **/
 public class CallbackInterceptor implements InstanceMethodsAroundInterceptor {
+
     @Override
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
-        MethodInterceptResult result) throws Throwable {
-        AbstractSpan abstractSpan = ContextManager.createLocalSpan("Producer/Callback");
+                             MethodInterceptResult result) throws Throwable {
+        RecordMetadata metadata = (RecordMetadata) allArguments[0];
+        AbstractSpan activeSpan = ContextManager.createLocalSpan("Kafka/Producer/Callback");
+        activeSpan.setComponent(ComponentsDefine.KAFKA_PRODUCER);
+        activeSpan.setLayer(SpanLayer.MQ);
```

Comment:
Usually, we don't set the layer to local span. Layer is in RPC process.

Reply:
Sorry, I am not sure about this part at that time, I will modify it